### PR TITLE
CNV-23832: Storage class dropdown when no defined in template or default in cluster

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

When there's no Storage class defined in the template or no default virt SC/ cluster default SC, trying to create a VM will stuck on provisioning. To avoid this we add a dropdown to in the create VM drawer menu on these cases.

## 🎥 Demo

After:
sc-required

### CODE REVIEW

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/hooks/useDefaultVMSource.ts [hunk 1]
```diff
@@ -20,9 +21,20 @@ const useDefaultVMSource = (vm: V1VirtualMachine) => {
...
+      ...(currentDiskSource as V1beta1DataVolumeSpec),
+    };
+
+    return isEqualObject(restDefaultSpec, restCurrentSpec);
+  };
+
   return {
     currentDiskSource,
-    isDefaultDiskSource: currentDiskSource === defaultDiskSource.current,
+    isDefaultDiskSource: isDefaultDiskSource(),
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/hooks/useDefaultVMSource.ts:37]: why function if u simply return the result?

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/StorageClassSelect/StorageClassSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        maxHeight={200}
+        onToggle={setIsOpenStorageClass}
+        placeholderText={t('Select storage class')}
+        selections={storageClassName}
+      >
+        {sortedStorageClasses?.map((name) => (
+          <SelectOption key={name} value={name} />
+        ))}
+      </Select>
+    </FormGroup>
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/StorageClassSelect/StorageClassSelect.tsx:51]: maybe I'm missing but where do u set the value of the dropdown to template default and, if not exist, to virt default and, if not exist, to cluster default, and if not exist, to first find?

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/StorageClassSelect/StorageClassSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [{ storageClasses }] = useDefaultStorageClass();
+
+  const [isOpenStorageClass, setIsOpenStorageClass] = useState<boolean>(false);
+
+  const sortedStorageClasses = useMemo(
+    () => storageClasses?.map((sc) => sc?.metadata?.name)?.sort(),
+    [storageClasses],
+  );
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/StorageClassSelect/StorageClassSelect.tsx:26]: maybe move this into useDefaultStorageClass?

Comment (avivtur) [reply src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/StorageClassSelect/StorageClassSelect.tsx:26]: Good idea, done thanks :)

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/hooks/useDefaultVMSource.ts (hunk 1 above)
Comment (avivtur) [reply src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/hooks/useDefaultVMSource.ts]: No good reason, changed thanks :)

**Pull request merged**